packages/edit-post/src/components/init-pattern-modal/index.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect, useDispatch } from '@wordpress/data';
+import { __, _x } from '@wordpress/i18n';
+import {
+	Modal,
+	Button,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+	ToggleControl,
+	TextControl,
+} from '@wordpress/components';
+import { useEffect, useState } from '@wordpress/element';
+import { store as editorStore } from '@wordpress/editor';
+import { privateApis as blockEditorPrivateApis } from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+
+import { unlock } from '../../lock-unlock';
+
+const { ReusableBlocksRenameHint } = unlock( blockEditorPrivateApis );
+
+export default function InitPatternModal() {
```

Comment:
I moved this in `edit-post` since this is the only place is used. Additionally we should probably look into how we can consolidate this one with `CreatePatternModal` from `patterns` package. This could be in a follow up though and not a block for this PR to land for 6.5